frontend/src/hooks/use-auth-header.ts

```diff
@@ -0,0 +1,80 @@
+import { useCallback, useState } from "react";
+import { useAccount, useSignMessage } from "wagmi";
+import { getToken, saveToken, clearToken, isTokenValid } from "@/lib/utils/jwt";
+import { useGetNonce } from "./profiles/use-get-nonce";
+import { generateSiweMessage } from "@/lib/utils/siwe";
+import { API_BASE_URL } from "@/lib/constants/apiConstants";
+
+interface AuthHeaders {
+    [key: string]: string;
+}
+
+export function useAuthHeader() {
```

Comment:
This hook isn't used anywhere. I actually like the underlying logic of triggering jwt sign in if token is expired or absent, so I think we should use this in the CRUD hooks for profile